library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java

```diff
@@ -375,11 +374,11 @@ public void run()
             lastScannedInfo = null;
             numDiscoveredPerIteration = 0;
             for (String afile : files) {
-              String filePath = new File(afile).getAbsolutePath();
-              LOG.debug("Scan started for input {}", filePath);
+              Path filePath = new Path(afile);
+              LOG.debug("Scan started for input {}", filePath.toString());
               Map<String, Long> lastModifiedTimesForInputDir;
-              lastModifiedTimesForInputDir = referenceTimes.get(filePath);
-              scan(new Path(afile), null, lastModifiedTimesForInputDir);
+              lastModifiedTimesForInputDir = referenceTimes.get(fs.getFileStatus(filePath).getPath().toString());
```

Comment:
No. In case of LocalFileSystem, if the filePath consists of relative path then again the keys are different. 
Setting the directory path in createScannedFileInfo() and accessing the value from referenceTimes must be sync.